Lib/test/test_tarfile.py

```diff
@@ -694,10 +694,11 @@ def test_extract_hardlink(self):
     def test_extractall(self):
         # Test if extractall() correctly restores directory permissions
         # and times (see issue1735).
-        tar = tarfile.open(tarname, encoding="iso8859-1")
-        DIR = os.path.join(TEMPDIR, "extractall")
-        os.mkdir(DIR)
-        try:
+        DIR = pathlib.Path(TEMPDIR) / "extractall"
+        with (
+            os_helper.temp_dir(DIR),
+            tarfile.open(tarname, encoding="iso8859-1") as tar
+        ):
```

Comment:
If it's not related to the PR itself, I think you can just revert this change (that way, your diff is smaller).